src/inspector/inspector.ts

```diff
@@ -1,55 +1,98 @@
 import { ZodError } from 'zod';
 import { VC, VCSchema } from './credentialSchemas/verifiableCredential';
 import { CalculatedAttributes, calculateAttributes } from './calculatedAttributes/calculateAttributes';
+import { ReasonedError, Result } from './calculatedAttributes/errors';
+import * as jose from 'jose';
+import { JWTPayload } from 'jose';
 
 export default function inspect(credential: string): InspectionResult {
-  try {
-    const parsedJson = JSON.parse(credential);
-    const parsedSchema = VCSchema.safeParse(parsedJson);
-
-    if (!parsedSchema.success) {
-      return {
-        type: 'InvalidCredential',
-        parsedJson,
-        error: parsedSchema.error,
-      };
-    }
+  const parsedJson = credentialToJSON(credential);
+  if (parsedJson.kind == 'error') {
+    return {
+      type: 'ParseError',
+      error: parsedJson.error,
+    };
+  }
 
+  const parsedSchema = VCSchema.safeParse(parsedJson.value.payload);
+
+  if (!parsedSchema.success) {
     return {
-      type: 'ValidCredential',
-      parsedJson: parsedJson as VC,
-      calculatedAttributes: calculateAttributes(parsedSchema.data),
+      type: 'InvalidCredential',
+      parsedJson: parsedJson.value,
+      error: parsedSchema.error,
     };
+  }
+
+  return {
+    type: 'ValidCredential',
+    parsedJson: parsedSchema.data,
+    calculatedAttributes: calculateAttributes(parsedSchema.data),
+  };
+}
+
+type ParsedJWT = {
+  type: 'JWT';
+  jwtPayload: JWTPayload;
+  payload: JSON;
+};
+
+type ParsedJson = {
+  type: 'JSON';
+  payload: JSON;
+};
+
+type ParsedCredential = ParsedJson | ParsedJWT;
+
+function credentialToJSON(credential: string): Result<ParsedCredential> {
```

Comment:
Would want some tests for this to make sure updates to this function by extra dependencies does not change its core functionality.